### PR TITLE
fix: eslint path

### LINKED ISSUE #282: Incorrect eslintPkgPath: `.../api.js/package.json`
- Operating System: Mac OS
- Node Version: 10.15.1
- NPM Version: 6.4.1
- webpack Version: 4.28.4
- eslint-loader Version: 2.2.0

### Expected Behavior

Expecting regular Vue builds to succeed.

### Actual Behavior

See https://github.com/vuejs/vue-cli/issues/4231

Vue builds are failing due to an eslint-loader error.

### Code

[This addition](https://github.com/webpack-contrib/eslint-loader/commit/b6c3938aae61c0390ac8d941af50cf36df14cfc3#diff-168726dbe96b3ce427e7fedce31bb0bcR167) seems to result in `eslintPkgPath` sometimes adding `/package.json` to the path in situations where it should not:
```js
  var eslintPkgPath = "eslint/package.json";
  if (userEslintPath) {
    eslintPkgPath = userEslintPath + "/package.json";
  }

  var eslintVersion = require(eslintPkgPath).version;
```

### How Do We Reproduce?
See https://github.com/vuejs/vue-cli/issues/4231

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Resolve #282

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -160,21 +160,12 @@ module.exports = function(input, map) {
...
-
-  var eslintVersion = require(eslintPkgPath).version;
-
   var config = assign(
     // loader defaults
     {
       cacheIdentifier: JSON.stringify({
         "eslint-loader": pkg.version,
-        eslint: eslintVersion
+        eslint: require(userOptions.eslintPath || "eslint").version
```
Comment (evilebottnawi) [inline index.js]: Let's do this more safe - move this above and save in variable and put this in `try/catch` and use `eslintValue || 'unknown version'`

Comment (evilebottnawi) [reply index.js]: Also old eslint version has `version`? Maybe we need more complex way to check this

Comment (ricardogobbosouza) [reply index.js]: From version >= 3

Comment (ricardogobbosouza) [reply index.js]: > Let's do this more safe - move this above and save in variable and put this in `try/catch` and use `eslintValue || 'unknown version'`

I will improve

Comment (evilebottnawi) [reply index.js]: Looks we break old version https://github.com/webpack-contrib/eslint-loader/blob/master/package.json#L19

Comment (ricardogobbosouza) [reply index.js]: Looks now

Comment (sodatea) [reply index.js]: I don't think any version of ESLint has the `version` field. It's `.CLIEngine.version` since v3, and v2 returns `undefined`.

**Pull request merged**